File: Java/007_Reverse Integer.java
class Solution {
    public int reverse(int x) {
        boolean sign = x >= 0;
        long tmp = (long)x;
        tmp = Math.abs(tmp);
        long res = 0;
        while (tmp > 0) {
            long digit = tmp % 10;
            tmp /= 10;
            res = res * 10 + digit;
        }
        if (sign) {
            if (res > Integer.MAX_VALUE)
                return 0;
            return (int)res;
        }
        else {
            if (-res < Integer.MIN_VALUE)
                return 0;
            return (int)-res;
        }
    }
}

public class Test {
    public static void main(String[] args) {
        Solution s = new Solution();
        System.out.println(s.reverse(123));
        System.out.println(s.reverse(-123));
    }
}

File: Python/007_Reverse Integer.py
class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        if x == 0:
            return 0
        flag = True if x < 0 else False
        res, x = 0, abs(x)
        while x != 0:
            res = res * 10 + x % 10
            x /= 10
        if flag and -res < -2147483648:
            res = 0
        if not flag and res > 2147483647:
            res = 0
        return -res if flag else res